x-pack/plugins/security_solution/public/common/hooks/use_update_execution_context.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useKibana } from '../lib/kibana';
+
+export const useUpdateExecutionContext = () => {
+  const { pathname } = useLocation();
+  const { executionContext } = useKibana().services;
+
+  useEffect(() => {
+    // setTimeout is required to ensure that EBT telemetry for the previous page is shipped before the new page is updated
+    setTimeout(() => {
+      executionContext.set({ page: pathname });
+    });
```

Comment:
nit: `setTimeout` without a time, isn't it `setImmediate`?

Reply:
They are not the same thing, but both push the function execution to the next event loop. So both should work here.